terraform/infrastructure/modules/05-cloudfront/04-cdn.tf

```diff
@@ -0,0 +1,239 @@
+resource "aws_cloudfront_origin_request_policy" "websocket" {
+  name    = "${var.name}-cloudfront-websockets"
+  comment = "Websocket Header Allow Policy"
+
+  cookies_config {
+    cookie_behavior = "none"
+  }
+
+  query_strings_config {
+    query_string_behavior = "all"
+  }
+
+  headers_config {
+    header_behavior = "whitelist"
+    headers {
+      items = [
+        "Sec-WebSocket-Key",
+        "Sec-WebSocket-Version",
+        "Sec-WebSocket-Protocol",
+        "Sec-WebSocket-Accept"
+      ]
+    }
+  }
+}
+
+module "cloudfront" {
+  source = "terraform-aws-modules/cloudfront/aws"
+
+  aliases = [var.domain_name]
+
+  enabled         = true
+  staging         = false
+  http_version    = "http2and3"
+  is_ipv6_enabled = true
+
+  # make it a bit cheaper. other options are PriceClass_200 and PriceClass_All
+  price_class      = "PriceClass_100"
+  retain_on_delete = false
+
+  # deployment for cloudfront usually takes a while
+  wait_for_deployment = false
+
+  # When you enable additional metrics for a distribution, CloudFront sends up to 8 metrics to CloudWatch in the US East (N. Virginia) Region.
+  # This rate is charged only once per month, per metric (up to 8 metrics per distribution).
+  create_monitoring_subscription = true
+
+  create_origin_access_identity = false
+
+  create_origin_access_control = true
+  origin_access_control = {
+    # we need to allow cloudfront to access S3 buckets
+    s3_oac = {
+      description      = "CloudFront S3 Origin Access Control"
+      origin_type      = "s3"
+      signing_behavior = "always"
+      signing_protocol = "sigv4"
+    }
+  }
+
+  # logging is disabled for now
+  # logging_config = {
+  #   bucket = module.log_bucket.s3_bucket_bucket_domain_name
+  #   prefix = "cloudfront"
+  # }
+
+  origin = {
+    frontend = {
+      domain_name           = var.frontend_domain_name
+      origin_access_control = "s3_oac" # key in `origin_access_control` map
+    }
+
+    scratchapp = {
+      domain_name           = var.scratchapp_domain_name
+      origin_access_control = "s3_oac"
+    }
+
+    backend = {
+      domain_name = var.backend_domain_name
+      custom_origin_config = {
+        http_port              = 80
+        origin_protocol_policy = "http-only"
+
+        https_port           = 443
+        origin_ssl_protocols = ["TLSv1.2"]
+      }
+    }
+  }
+
+  default_cache_behavior = {
+    target_origin_id       = "frontend" # key in `origin` map
+    viewer_protocol_policy = "redirect-to-https"
+    allowed_methods        = ["GET", "HEAD", "OPTIONS"]
+    cached_methods         = ["GET", "HEAD", "OPTIONS"]
+
+    use_forwarded_values = false
+
+    # disable caching
+    cache_policy_id = "4135ea2d-6df8-44a3-9df3-4b5a84be39ad"
+
+    # https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-managed-response-headers-policies.html#managed-response-headers-policies-security
+    response_headers_policy_id = "67f7725c-6f97-4210-82d7-5512b31e9d03"
```

Comment:
these numbers are magic.
as discussed, it might make sense to use a data resource to retrieve them from the AWS provider.